lib/python/ray/services.py

```diff
@@ -12,17 +12,24 @@
 import subprocess
 import sys
 import time
-from collections import namedtuple
+from collections import namedtuple, OrderedDict
 
 # Ray modules
 import photon
 import plasma
 import global_scheduler
 
-# all_processes is a list of the scheduler, object store, and worker processes
-# that have been started by this services module if Ray is being used in local
-# mode.
-all_processes = []
+# This is a dictionary tracking all of the processes of different types that
+# have been started by this services module. Note that the order of the keys is
+# important because it determines the order in which these processes will be
+# terminated when Ray exits, and certain orders will cause errors to be logged
+# to the screen.
+all_processes = OrderedDict([("workers", []),
+                             ("local_schedulers", []),
+                             ("plasma_managers", []),
+                             ("plasma_stores", []),
+                             ("global_schedulers", []),
+                             ("redis_servers", [])])
```

Comment:
these string constants appear in a few places in the code. I think it might help to define them in the enum-like style in one place. E.g., 
```
PROCTYPE_WORKER="workers"
PROCTYPE_LS="local_schedulers"
PROCTYPE_PLASMA_MANAGER=...
PROCTYPE_GS=
PROCTYPE_REDIS_SERVER="redis_servers"
```